File: library/src/main/java/com/github/lany192/box/mvp/ListPageView.java
package com.github.lany192.box.mvp;

import java.util.List;

/**
 * 分页列表页视图
 */
public interface ListPageView<T> extends BaseView {

    void addItems(List<T> items);

    void fail();

    void stop();

    void end();

    void gotoTop();

    List<T> getItems();

    void setItems(List<T> items);
}

File: library/src/main/java/com/github/lany192/box/mvp/ListPageView.kt
package com.github.lany192.box.mvp

/**
 * 分页列表页视图
 */
interface ListPageView<T> : BaseView {
    fun addItems(items: List<T>?)
    fun fail()
    fun stop()
    fun end()
    fun gotoTop()
    var items: List<T>?
}